soccer/Context.hpp

```diff
@@ -22,4 +25,9 @@ struct Context {
 
     std::vector<std::unique_ptr<VisionPacket>> vision_packets;
     WorldState world_state;
+
+    // construct these right away so they can be used from the SystemState
+    // constructor
```

Comment:
If you want them to be constructed before the systemstate constructor they should be placed before the systemstate in the class declaration (so move this line up)